### ISSUE TITLE
Unable to shift focus from TextField, stuck in editing mode

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#53065]
Internal b/150663666

Observe that when using TextField or TextFormField in Flutter on iOS platform, users cannot shift Accessibility Focus in VoiceOver once they're in editing mode.

Background: One of the criteria for A11y testing is that users should be able to swipe Left / Right / Up / Down gestures in VoiceOver mode regardless where they are.

Actual result: Users are stuck in TextField / TextFormField editing mode.

Steps to reproduce:
1. Turn on VoiceOver
2a. Navigate to a screen that has a TextField or TextFormField
(OR 2b. Navigate to a screen that has a TextField or TextFormField with autofocus: true)
3. Navigate to the TextField using swipe gestures
4. Double to focus on the TextField
5. Observe that swiping left / right / up / down doesn't shift a11y anymore

**Comment by goderbauer** [comment]
My understanding of the problem described in this bug is that a textfield loses a11y focus when it gains input focus. Since nothing has a11y focus, linear navigation (swiping left/right) doesn't work anymore. You can touch anywhere on the screen to move the a11y focus there and after that linear navigation works again.

Native textfields keep their a11y focus when they gain input focus. So, this is definitely a bug in Flutter. 

On the framework side I verified that textfield nodes keep the same node id whether they have input focus or not. It's not clear to my why iOS decides to remove the a11y focus. Maybe the iOS accessibility bridge is accidentally creating a new SemanticsNode when the textfield receives focus?

@xster Can somebody from the platforms team investigate this?

**Comment by xster** [comment]
@goderbauer sounds like this is just a new thing being tested (and may have never worked) or is it a regression?

**Comment by goderbauer** [comment]
I can't say with certainty whether this is a regression or not since there's no testing for the a11y bridge. But either way, the current behavior is buggy.

**Comment by xster** [comment]
Seems like a side effect of us bringing up the keyboard. Once the keyboard is present, going through the same repro process doesn't break the accessibility focus.

**Comment by xster** [comment]
There were a lot of pieces involved so it took me a while to read the code, but the solution turned out simpler than I thought. We have a double UIAccessibilityElements tree, one in our SemanticsObjects tree and a parallel one in our UIView hierarchy tree. 

When a Flutter text field turns active, the corresponding UIAccessibilityElements in our SemanticsObjects tree (TextInputSemanticsObject) flips its delegate to a real offscreen UIView which also implements UIAccessibilityElements (the FlutterTextInputView which is a UIView <UITextInput> (which implements accessibility)). 

Forwarding various accessibility values from the real UIView makes sense to me but I'm not sure whether we should forward all the accessibility events to the real UIView, like `accessibilityActivate` to the UIView. 

Either case, since that UIView is probably higher in the traversal, it gets the focus instead. Since it's on a different tree, it probably doesn't see the SemanticsObjects as its sibling. 

Setting `accessibilityElementsHidden` to `YES` or setting `isAccessibilityElement` to `NO` on `FlutterTextInputView` fixes the issue. Tangentially, there's also a another indirection via `FlutterTextInputViewAccessibilityHider`. That's not necessary. `FlutterTextInputView` is-a UIView. We can just set `accessibilityElementsHidden` to `YES` on it directly. 

If it's pressing, the user can upstream the fix directly. It'll probably take me a while to write an integration test for a PR.
